core/block_svg.ts

```diff
@@ -245,19 +245,16 @@ export class BlockSvg
 
   /** Selects this block. Highlights the block visually. */
   select() {
-    if (this.isShadow()) {
-      this.getParent()?.select();
+    if (this.isShadow() && this.getParent()) {
+      // Shadow blocks should not be selected.
+      this.getParent()!.select();
       return;
     }
     this.addSelect();
   }
 
   /** Unselects this block. Unhighlights the blockv visually.   */
   unselect() {
-    if (this.isShadow()) {
-      this.getParent()?.unselect();
-      return;
-    }
```

Comment:
Why was this code just deleted, and not replaced with:

```JS
    if (this.isShadow() && this.getParent()) {
      // Shadow blocks should not be selected.
      this.getParent()!.unselect();
      return;  // Or maybe fall through to this.removeSelect()?
    }
```

Doesn't the code as proposed in this PR create a seemingly-buggy asymmetry where where calling `select` and then `unselect` on a shadow block will leave the parent block selected?